tests/TestCase/Mailer/EmailTest.php

```diff
@@ -1588,6 +1588,63 @@ public function testSendWithNoContentDispositionAttachments()
         $this->assertContains('--' . $boundary . '--', $result['message']);
     }
 
+    /**
+     * Test an attachment filename with non-ASCII characters.
+     *
+     * @return void
+     */
+    public function testSendWithNonAsciiFilenameAttachments()
+    {
+        $this->Email->setTransport('debug');
+        $this->Email->setFrom('cake@cakephp.org');
+        $this->Email->setTo('cake@cakephp.org');
+        $this->Email->setSubject('My title');
+        $this->Email->setEmailFormat('both');
+        $this->Email->setAttachments([
+            'gâteau.png' => [
+                'file' => CORE_PATH . 'VERSION.txt',
+                'contentId' => 'abc123'
+            ]
+        ]);
+        $result = $this->Email->send('Hello');
+
+        $boundary = $this->Email->getBoundary();
+        $this->assertContains('Content-Type: multipart/mixed; boundary="' . $boundary . '"', $result['headers']);
+        $expected = "--$boundary\r\n" .
+            "Content-Type: multipart/related; boundary=\"rel-$boundary\"\r\n" .
+            "\r\n" .
+            "--rel-$boundary\r\n" .
+            "Content-Type: multipart/alternative; boundary=\"alt-$boundary\"\r\n" .
+            "\r\n" .
+            "--alt-$boundary\r\n" .
+            "Content-Type: text/plain; charset=UTF-8\r\n" .
+            "Content-Transfer-Encoding: 8bit\r\n" .
+            "\r\n" .
+            'Hello' .
+            "\r\n" .
+            "\r\n" .
+            "\r\n" .
+            "--alt-$boundary\r\n" .
+            "Content-Type: text/html; charset=UTF-8\r\n" .
+            "Content-Transfer-Encoding: 8bit\r\n" .
+            "\r\n" .
+            'Hello' .
+            "\r\n" .
+            "\r\n" .
+            "\r\n" .
+            "--alt-{$boundary}--\r\n" .
+            "\r\n" .
+            "--rel-$boundary\r\n" .
+            "Content-Disposition: inline; filename=\"gteau.png\"; filename*=utf-8''g%C3%A2teau.png\r\n" .
```

Comment:
Is having the filename defined twice correct? I couldn't find any examples in the RFC that included the same parameter multiple times.

Reply:
There's a subtle difference `filename` vs `filename*`.

Reply:
When both "filename" and "filename*" are present in a single header field value, "filename*" is preferred over "filename" when both are understood.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition

Reply:
@garas Yes in http headers. Is that also true in multipart email messages?

Reply:
According to NodeMailer, [yes](https://blog.nodemailer.com/2017/01/27/the-mess-that-is-attachment-filenames/). They indicate that the filename parameter (without the asterisk) cannot be encoded and should only include Latin characters (any other characters should be stripped or trans-coded to Latin equivalent). The filename* parameter (with the asterisk) is supposed to be URL parameter encoded to allow for non-Latin characters -- though not all clients may recognize or even use either of the filename fields. If you really want to support older clients you would also need to include the "name" parameter encoded (using encoded-word) as part of the Content-Type header (e.g. the pre-1997 standard).

Reply:
To be fair, it doesn't say anything about fallbacks, and [it seems to generate only the encoded variant for unicode filenames](https://github.com/nodemailer/nodemailer/blob/v6.3.1/test/mime-node/mime-node-test.js#L591-L604). Looking at for example Mozilla's stuff (ie Thunderbird, etc.) [it seems to support it](https://dxr.mozilla.org/comm-central/rev/2a29ee0adb310b54a6a2df72034953fed8f2b043/comm/mailnews/mime/jsmime/test/test_header.js#45), but that doesn't mean too much of course.

Theoretically unrecognized parameters should be ignored according to https://tools.ietf.org/html/rfc2183#section-2.8, but I couldn't find anything specific about fallbacks for encoded parameter values in general. But as stated earlier, you can't necessarily rely on RFCs anyways, so this probably either needs some real world testing, or the fallback needs to go and people need to make their filenames ASCII-only in beforehand if they need maximum compatibility and the "old" parameter style.